robottelo/vm.py

```diff
@@ -232,19 +235,46 @@ def create(self):
             ping_from_hostname = settings.server.hostname
 
         # Give some time to machine boot
-        result = ssh.command(
-            u'for i in {{1..60}}; do ping -c1 {0}.local && exit 0; sleep 1;'
-            u' done; exit 1'.format(self._target_image),
-            ping_from_hostname,
-            connection_timeout=30
-        )
-        if result.return_code != 0:
-            logger.error('Failed to obtain VM IP, reverting changes')
-            self.destroy()
-            raise VirtualMachineError(
-                'Failed to fetch virtual machine IP address information')
-        output = ''.join(result.stdout)
-        self.ip_addr = output.split('(')[1].split(')')[0]
+        for i in range(60):
+            qemu_ga_check = ssh.command(
+                u'virsh qemu-agent-command {0} '
+                '\'{{"execute":"guest-network-get-interfaces"}}\''.format(self.hostname),
+                ping_from_hostname,
+                connection_timeout=30
+            )
+            if qemu_ga_check.return_code != 0:
+                if 'guest agent is not connected' in qemu_ga_check.stderr:
+                    # this means the agent wasn't started yet (vm still booting)
+                    sleep(1)
+                else:
+                    # this means there's another error with agent, e.g. it is not configured
+                    break
+            else:
+                ifaces = json.loads(qemu_ga_check.stdout[0])
+                mgmt_if = [i for i in ifaces['return']
+                           if 'eth0' in i['name'] or 'ens3' in i['name']]
+                if len(mgmt_if) > 0 and mgmt_if[0].get('ip-addresses'):
```

Comment:
This is tricky.
If there were both, but the first didn't have the ip address, we still couldn't tell which one is the right one, as it would still be possible, that the 1st interface does not have the IP address YET.
- we should develop some mechanism to correctly tell which one of the interfaces is the management one.
Anyway, your suggestion makes sense. I'll try to implement it

Reply:
```I'll try to implement it``` <- in this PR or in future

I'm for future since this PR just introduced qemu-ga IP discovery and later on we can enhance its logic further

Reply:
@mirzal @lpramuk - in fact I'll go further and pick the interface with the MAC address returned by snap-guest call.
- In future we want to update snap-guest to allow adding multiple interfaces and returning the MGMT_MAC.